### PR TITLE
DMX support for video instream and more user id support

### PR DESCRIPTION
**Pull request by stevealliance**
## Type of change

- [x] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
Update DMX adapter to support video 
Bug on user id now corrected
Test with prebid version 4.0+

### CODE REVIEW

File: modules/districtmDMXBidAdapter.js [hunk 1]
```diff
@@ -122,14 +147,34 @@ export const spec = {
...
       obj.secure = 1;
-      obj.banner = {
-        topframe: 1,
-        w: cleanSizes(dmx.sizes, 'w'),
-        h: cleanSizes(dmx.sizes, 'h'),
-        format: cleanSizes(dmx.sizes).map(s => {
-          return {w: s[0], h: s[1]};
-        }).filter(obj => typeof obj.w === 'number' && typeof obj.h === 'number')
-      };
+      obj.bidfloor = dmx.params.floor || 0;
```
Comment (patmmccann) [inline modules/districtmDMXBidAdapter.js:150]: Please support the floors module as well as your own bidder params. See https://github.com/prebid/Prebid.js/pull/5329/files for an example

File: modules/districtmDMXBidAdapter.js [hunk 2]
```diff
@@ -83,12 +100,20 @@ export const spec = {
...
     let eids = [];
-    if (bidRequest && bidRequest.userId) {
-      bindUserId(eids, utils.deepAccess(bidRequest, `userId.idl_env`), 'liveramp.com', 1);
-      bindUserId(eids, utils.deepAccess(bidRequest, `userId.digitrustid.data.id`), 'digitru.st', 1);
-      bindUserId(eids, utils.deepAccess(bidRequest, `userId.id5id`), 'id5-sync.com', 1);
-      bindUserId(eids, utils.deepAccess(bidRequest, `userId.pubcid`), 'pubcid.org', 1);
-      bindUserId(eids, utils.deepAccess(bidRequest, `userId.tdid`), 'adserver.org', 1);
+    if (bidRequest[0] && bidRequest[0].userId) {
+      bindUserId(eids, utils.deepAccess(bidRequest[0], `userId.idl_env`), 'liveramp.com', 1);
+      bindUserId(eids, utils.deepAccess(bidRequest[0], `userId.digitrustid.data.id`), 'digitru.st', 1);
```
Comment (patmmccann) [inline modules/districtmDMXBidAdapter.js:105]: digitrust is deprecated

Comment (stevealliance) [reply modules/districtmDMXBidAdapter.js]: user id was removed

File: modules/districtmDMXBidAdapter.js (hunk 1 above)
Comment (stevealliance) [reply modules/districtmDMXBidAdapter.js]: support for floor module was added

File: modules/districtmDMXBidAdapter.js [hunk 3]
```diff
@@ -310,4 +375,65 @@ export function bindUserId(eids, value, source, atype) {
...
+    {key: 'VPAID_1_0', value: 1},
+    {key: 'VPAID_2_0', value: 2},
+    {key: 'MRAID_1', value: 3},
+    {key: 'ORMMA', value: 4},
+    {key: 'MRAID_2', value: 5},
+    {key: 'MRAID_3', value: 6},
+  ];
+  if (protocols) {
+    return listProtocols.filter(p => {
+      return protocols.includes(p.key);
```
Comment (patmmccann) [inline modules/districtmDMXBidAdapter.js:391]: tests fail on ie 11 bc https://caniuse.com/#search=includes

File: modules/districtmDMXBidAdapter.js [hunk 4]
```diff
@@ -27,9 +39,14 @@ export const spec = {
...
+              nBid.mediaType = bid.mediaTypes && bid.mediaTypes.video ? 'video' : null;
+              if (nBid.mediaType) {
+                nBid.vastXml = cleanVast(nBid.adm);
+              }
               if (nBid.dealid) {
                 nBid.dealId = nBid.dealid;
               }
+              nBid.uuid = nBid.bidId;
               nBid.ad = nBid.adm;
               nBid.netRevenue = true;
```
Comment (patmmccann) [inline modules/districtmDMXBidAdapter.js:51]: can you also write nBid.adomain to meta.advertiserDomains as per #3115 ? You can see a bunch of example commits doing this at the bottom of the issue. I just hit the dmx endpoinnt and saw adomain and cat populated as arrays.

File: modules/districtmDMXBidAdapter.js (hunk 3 above)
Comment (stevealliance) [reply modules/districtmDMXBidAdapter.js:391]: includes was changed for indexOf !== -1

File: modules/districtmDMXBidAdapter.js (hunk 4 above)
Comment (stevealliance) [reply modules/districtmDMXBidAdapter.js:51]: this was update

**Pull request merged**